wordpress/wp-content/plugins/sil-dictionary-webonary/webonary/Webonary_Cloud.php

```diff
@@ -76,139 +76,73 @@ private static function sematicDomainToLink($lang, $domain) {
+				if (preg_match_all(
+					'/(?:<span class=\"name\">|\G)+?(<span lang=\"(\S+)\">(.*?)<\/span>)/',
+					$semDom,
+					$semDomNames) > 0) {
+					// <span lang="en">Language and thought</span>
+					$newSemDom = $semDom;
+					foreach($semDomNames[1] as $index => $semDomNameSpan) {
+						$lang = $semDomNames[2][$index];
+						$domain = $semDomNames[3][$index];
+						// @todo: For some reason, only the first semantic domain is made  in a link. Need to verify if correct.
+						$newSemDom = str_replace(
+							$semDomNameSpan,
+							'<span lang="' . $lang . '">' . self::sematicDomainToLink($lang, $domain) . '</span>',
+							$newSemDom);
 					}
-					$displayXhtml = str_replace($semDom, $newSemDom, $displayXhtml);
-				}
-			}
-		}
-		else {
-			$mainHeadWord = '<span class="mainheadword"><span lang="' . $entry->mainHeadWord[0]->lang . '">'
-				. '<a href="' . get_site_url() . '/' . $id . '">' . $entry->mainHeadWord[0]->value . '</a></span></span>';
-					
-			$lexemeform = '';
-			if ($entry->audio->src != '') {
-				$lexemeform .= '<span class="lexemeform"><span><audio id="' . $entry->audio->id . '">';
-				$lexemeform .= '<source src="' . self::remoteFileUrl($dictionaryId . '/' . $entry->audio->src) . '"></audio>';
-				$lexemeform .= '<a class="' . $entry->audio->fileClass . '" href="#' . $entry->audio->id . '" onClick="document.getElementById(\'' . $entry->audio->id .   '\').play()"> </a></span></span>';
-			}
-		
-			// TODO: There can be multiple media files, e.g. Hayashi, one for lexemeform and another in pronunciations
-			$sharedgrammaticalinfo = '<span class="sharedgrammaticalinfo"><span class="morphosyntaxanalysis"><span class="partofspeech"><span lang="' . $entry->morphoSyntaxAnalysis->partOfSpeech[0]->lang . '">' . $entry->morphoSyntaxAnalysis->partOfSpeech[0]->value . '</span></span></span></span>';
-		
-			$sensecontent = '<span class="sensecontent"><span class="sense" entryguid="' . $id . '">'
-				. '<span class="definitionorgloss">';
-			foreach ($entry->senses[0]->definitionOrGloss as $definition)	{
-				$sensecontent .= '<span lang="' . $definition->lang . '">' . $definition->value . '</span>';
-			}
-			$sensecontent .= '</span></span>';
-		
-			$senses = '<span class="senses">' . $sharedgrammaticalinfo . $sensecontent . '</span>';
-		
-			$pictures = '';
-			if (count($entry->pictures)) {
-				$pictures = '<span class="pictures">';
-				foreach ($entry->pictures as $picture)	{
-					$pictureUrl = self::remoteFileUrl($dictionaryId . '/' . $picture->src);
-					$pictures .= '<div class="picture">';
-					$pictures .= '<a class="image" href="' . $pictureUrl . '">';
-					$pictures .= '<img src="' . $pictureUrl . '"></a>';
-					$pictures .= '<div class="captioncontent"><span class="headword"><span lang="' . $definition->lang . '">' . $picture->caption . '</span></span></div>';
-					$pictures .= '</div>';
 				}
-				$pictures .= '</span>';	
+				$displayXhtml = str_replace($semDom, $newSemDom, $displayXhtml);
```

Comment:
Looks like this line should be inside the `if` statement above it above it because that's where `$newSemDom` is defined.

Reply:
yes, correct. thank you. fixed.